pkg/apis/openebs.io/v1alpha1/cas_volume.go

```diff
@@ -174,23 +174,17 @@ type CASVolume struct {
 
 // CASVolumeSpec has the properties of a cas volume
 type CASVolumeSpec struct {
-	// Capacity will hold the capacity of this Volume
-	Capacity string `json:"capacity"`
-	// Iqn will hold the iqn value of this Volume
-	Iqn string `json:"iqn"`
-	// TargetPortal will hold the target portal for this volume
-	TargetPortal string `json:"targetPortal"`
-	// TargetIP will hold the targetIP for this Volume
-	TargetIP string `json:"targetIP"`
-	// TargetPort will hold the targetIP for this Volume
-	TargetPort string `json:"targetPort"`
+	// ISCSISpec will hold the ISCSI properties of this volume
+	ISCSISpec
```

Comment:
I think it will be better to have `ISCSIVOLUME` or `ISCSISOURCE` struct 
```
// Represents an ISCSI disk.
// ISCSI volumes can only be mounted as read/write once.
type ISCSIVolume struct {
	// Required: iSCSI target portal
	// the portal is either an IP or ip_addr:port if port is other than default (typically TCP ports 860 and 3260)
	// +optional
	TargetPortal string
	// Required:  target iSCSI Qualified Name
	// +optional
	IQN string
	// Required: iSCSI target lun number
	// +optional
	Lun int32
	// Filesystem type to mount.
	// Must be a filesystem type supported by the host operating system.
	// Ex. "ext4", "xfs", "ntfs". Implicitly inferred to be "ext4" if unspecified.
	// TODO: how do we prevent errors in the filesystem from compromising the machine
	// +optional
	FSType string
	// Optional: Defaults to false (read/write). ReadOnly here will force
	// the ReadOnly setting in VolumeMounts.	
```

Reply:
I think LUN could be moved to `ISCSISpec`, `FSType` is not required I guess. Please look at other comments explaining why I am following above structure.

Reply:
Above was the just the example ... we have to find out what is required and not....and arrange them in a proper design